banyand/metadata/client.go

```diff
@@ -35,17 +35,32 @@ import (
 
 const flagEtcdEndpointsName = "etcd-endpoints"
 
+const flagEtcdUsername = "etcd-username"
+
+const flagEtcdPassword = "etcd-password"
+
+const flagEtcdTLSCAFile = "etcd-tls-ca-file"
+
+const flagEtcdTLSCertFile = "etcd-tls-cert-file"
+
+const flagEtcdTLSKeyFile = "etcd-tls-key-file"
+
 // NewClient returns a new metadata client.
 func NewClient(_ context.Context) (Service, error) {
 	return &clientService{closer: run.NewCloser(1)}, nil
 }
 
 type clientService struct {
-	namespace      string
-	schemaRegistry schema.Registry
-	alc            *allocator
-	closer         *run.Closer
-	endpoints      []string
+	namespace       string
+	etcdUsername    string
+	etcdPassword    string
+	etcdTLSCAFile   string
+	etcdTLSCertFile string
+	etcdTLSKeyFile  string
```

Comment:
Both of these fields must either be present together or not at all. Please add a checker to verify it.